src/app/shared/dot-save-on-deactivate-service/save-on-deactivate.ts

```diff
@@ -0,0 +1,10 @@
+import { Observable } from 'rxjs/Observable';
+import { DotDialog } from '../models/dot-confirmation/dot-confirmation.model';
+import { ResponseView } from 'dotcms-js/dotcms-js';
+
+export interface OnSaveDeactivate {
+    isModelChanged(): boolean;
+    onDeactivateSave(): Observable<any>;
+    onDeactivateSaveError(response: ResponseView): void;
+    getSaveWarningMessages(): DotDialog;
```

Comment:
This is not returning a DotDialog, is just a couple of message, I will go with <any> for now. Will like to hear other opinions tho.